### PR TITLE
Improve Remote failure error messaging

### PR DESCRIPTION
**Pull request by gestrich**

### CODE REVIEW

File: LoopKit/Service/RemoteDataService.swift [hunk 1]
```diff
@@ -118,5 +118,16 @@ public extension RemoteDataService {
     var pumpEventDataLimit: Int? { return nil }
     var settingsDataLimit: Int? { return nil }
```
Comment (gestrich) [inline LoopKit/Service/RemoteDataService.swift:120]: Replace the signature with a Result type so I can capture the Error. I'm partial towards throwing APIs in Swift these days but I see Result types more common throughout Loop.

File: LoopKit/Notification/LoopNotificationCategory.swift [hunk 2]
```diff
@@ -20,6 +20,7 @@ public enum LoopNotificationCategory: String {
     case alert
     case remoteBolus
     case remoteBolusFailure
+    case remoteCommandExpired
```
Comment (ps2) [inline LoopKit/Notification/LoopNotificationCategory.swift:23]: This and the type above should probably go away when error reporting is being done correctly in the caregiver app.

File: LoopKit/Service/RemoteDataService.swift (hunk 1 above)
Comment (ps2) [reply LoopKit/Service/RemoteDataService.swift:120]: When the error path is correctly feeding errors to the caregiver app, the flow of errors here will be obviated as well.

**Review by ps2 — APPROVED** [review]
As much as I don't like adding more code that will eventually have to be rewound, I can understand this making debugging easier for now. I would hope some progress can be made towards the right path soon, as the more we invest in this workaround, the more work the detangling later is going to be.

**Pull request merged**